src/components/Collection/ListViewItem.tsx

```diff
@@ -45,17 +48,30 @@ const renderFileSources = (sources: SeriesSizesFileSourcesType): string => {
   return output.join(' | ');
 };
 
-const SeriesTag = ({ text, type }: { text: string, type: 'AniDB' | 'User' }) => (
-  <div
-    className={cx(
-      'text-xs font-semibold flex gap-x-2 items-center border-2 border-panel-tags rounded-lg p-2 whitespace-nowrap capitalize',
-      type === 'User' ? 'text-panel-text-important' : 'text-panel-text-primary',
-    )}
-  >
-    <Icon path={mdiTagTextOutline} size="1rem" />
-    <span className="text-panel-text">{text}</span>
-  </div>
-);
+const SeriesTag = React.memo(({ text, type }: { text: string, type: 'User' | 'AniDB' }) => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const handleClick = useEventCallback(() => {
+    navigate('/webui/collection', { state: { isTagLink: true } });
+    addFilterCriteriaToStore('HasTag').then(() => {
+      dispatch(setFilterTag({ HasTag: [{ Name: text, isExcluded: false }] }));
+    }).catch(console.error);
```

Comment:
```suggestion
    dispatch(setFilterTag({ HasTag: [{ Name: text, isExcluded: false }] }));
    addFilterCriteriaToStore('HasTag').catch(console.error);
```

This will fix the modal issue. We don't need to wait for the criteria data to add the condition